### PR TITLE
Add support for user selection of their AWS role in the case of multi…

### PR DESCRIPTION
**Pull request by nathan-v**
Add support for user selection of their AWS role in the case of mutiple roles. Addresses #4 

Also fixes a typo in main.py and an incorrect comment in aws_test.py

### CODE REVIEW

**Comment by nathan-v** [comment]
Example log output from the use of multiple role selection with --reup

```
17:10:16   (INFO) Nextdoor Okta Auther v0.0.1
Password: 
17:10:21   (INFO) Successfully authed Nathan V
17:10:21   (INFO) Getting SAML Assertion from my_awesome_okta
17:10:21   (WARNING) Multiple AWS roles found; please select one
[0] Role: arn:aws:iam::012345678910:role/admin_noiam
[1] Role: arn:aws:iam::012345678910:role/readonly
[2] Role: arn:aws:iam::012345678910:role/admin_full
Select a role from above: 2
17:10:22   (INFO) Assuming role: arn:aws:iam::012345678910:role/admin_full
17:10:23   (INFO) Wrote profile "default" to /home/nathan/.aws/credentials
17:10:23   (INFO) Session expires at 2018-01-23 02:10:23+00:00
17:10:23   (INFO) Reup enabled, sleeping...
18:00:31   (INFO) Getting SAML Assertion from my_awesome_okta
18:00:31   (INFO) Assuming role: arn:aws:iam::012345678910:role/admin_full
18:00:32   (INFO) Wrote profile "default" to /home/nathan/.aws/credentials
18:00:32   (INFO) Session expires at 2018-01-23 03:00:32+00:00
18:00:32   (INFO) Reup enabled, sleeping...
```

**Comment by nathan-v** [comment]
@diranged  ping. :)

**Comment by opragel** [comment]
@diranged This is like the code changes I made for multiple role support, but done right. I'd love for it to be merged..! 🙏

**Comment by diranged** [comment]
Sorry guys - I promise I'll review it this week. I looked briefly at it, but I am out on vacation right now. One quick thought before I dig too far into it.

Rather than prompting for the profile - what if we were to write out credentials for _all_ the profiles and keep them all in sync. So if you logged into Okta and were presented with 3 possible profiles ("admin", "user", "readonly"), we could populate the appropriate profiles in the .aws/credentials file so that the user can select them by using `AWS_PROFILE=...`.

**Comment by nathan-v** [comment]
I thought about doing something like that longer-term but my thought was adding support for a config file so that you could specify, in a more static way, how to handle multiple AWS accounts and/or roles since the command-line options to support all those roles could get complicated or mean having to make assumptions for the user on how to name the profiles.

It's definitely possible, though.

**Comment by opragel** [comment]
@diranged We have multiple AWS accounts hooked up to one endpoint in Okta, it's nice not having to assume anything about available roles in advance. I do think it would be very good to have a feature like that (being able to alias the role names would be nice), but I'd prefer for the interactive option to remain as the default if multiple roles are available but one isn't specified in advance as an argument.

@nathan-v Thanks for this, been using your fork since it was pushed! Works great.

File: nd_okta_auth/aws.py [hunk 1]
```diff
@@ -160,16 +176,22 @@ def assume_role(self):
...
-            role = self.assertion.roles()[0]
-        except xml.etree.ElementTree.ParseError:
-            log.error('Could not find any Role in the SAML assertion')
-            log.error(self.assertion.__dict__)
-            raise InvalidSaml()
+        if self.role is None:
+            try:
+                if len(self.assertion.roles()) > 1:
+                    raise MultipleRoles
+                else:
```
Comment (diranged) [inline nd_okta_auth/aws.py]: Style nit - no need for the `else:` block here.. 

```py
if len(self.assertion.roles()) > 1:
    raise MultipleRoles
self.role = self.assertion.roles()[0]
```

File: nd_okta_auth/main.py [hunk 2]
```diff
@@ -166,17 +166,27 @@ def main(argv):
...
             time.sleep(15)
             continue
 
-        log.info('Getting SAML Assertion from {org}'.format(
-            org=config.org))
-
-        try:
+        # Only set up the session if it's still None
+        if session is None:
             assertion = okta_client.get_assertion(appid=config.appid,
```
Comment (diranged) [inline nd_okta_auth/main.py]: This should stay wrapped in the try/except block - it makes an outbound web call with the sessions library (https://github.com/Nextdoor/nd_okta_auth/blob/master/nd_okta_auth/okta.py#L266-L267).

**Comment by nathan-v** [comment]
@diranged  Sorry for the delay; the changes have been added and PR updated.

**Comment by diranged** [comment]
@nathan-v , @opragel 

Have either of you run into this? I just started using the new ocde today, and I've seen this 3-4 times so far:

```
14:15:55   (INFO) Successfully authed Matt Wise
14:15:57   (INFO) Getting SAML Assertion from nextdoor
14:15:57   (INFO) Assuming role: arn:aws:iam::xxx:role/xxx
14:15:57   (INFO) Wrote profile "eng" to /Users/diranged/.aws/credentials
14:15:57   (INFO) Session expires at 2018-03-08 23:15:58+00:00
14:15:57   (INFO) Reup enabled, sleeping...
15:06:04   (INFO) Getting SAML Assertion from nextdoor
15:06:04   (INFO) Assuming role: arn:aws:iam::xxx:role/xxx
Traceback (most recent call last):
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/bin/nd_okta_auth", line 11, in <module>
    load_entry_point('nd-okta-auth==0.0.1', 'console_scripts', 'nd_okta_auth')()
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/lib/python2.7/site-packages/nd_okta_auth/main.py", line 207, in entry_point
    raise SystemExit(main(sys.argv))
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/lib/python2.7/site-packages/nd_okta_auth/main.py", line 182, in main
    session.assume_role()
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/lib/python2.7/site-packages/nd_okta_auth/aws.py", line 194, in assume_role
    SAMLAssertion=self.assertion.encode())
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/lib/python2.7/site-packages/botocore/client.py", line 314, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/diranged/git/git.corp.nextdoor.com/dotfiles/.venv-darwin/lib/python2.7/site-packages/botocore/client.py", line 612, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.ExpiredTokenException: An error occurred (ExpiredTokenException) when calling the AssumeRoleWithSAML operation: Token must be redeemed within 5 minutes of issuance
MacBook-Pro-6:dotfiles diranged$ 
```

**Comment by nathan-v** [comment]
@diranged  I haven't yet but I'll try to repro today.

**Pull request merged**